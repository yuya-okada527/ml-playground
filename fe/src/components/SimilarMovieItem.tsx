/* eslint-disable jsx-a11y/anchor-is-valid */
// TODO 暫定対応
import {
  createStyles,
  Grid,
  ListItem,
  makeStyles,
  Paper,
  Theme,
  Typography,
} from "@material-ui/core";
import Link from "next/link";
import Image from "material-ui-image";
import React from "react";
import { Movie } from "../interfaces";
import { makeTitle } from "./MovieDetail";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    link: {
      textDecoration: "none",
    },
    similarMovieItem: {
      paddingLeft: 0,
      paddingRight: 0,
      paddingBottom: theme.spacing(1),
      paddingTop: theme.spacing(0),
    },
    similarMovieItemPaper: {
      width: "100%",
      transition: "0.3s",
      "&:hover": {
        backgroundColor: "#eee",
        opacity: 0.8,
      },
    },
    similarMovieTitle: {
      margin: theme.spacing(1),
    },
  })
);

type SimilarMovieItemProps = {
  movie: Movie;
};

/**
 * 類似映画アイテムを描画する
 */
const SimilarMovieItem: React.FC<SimilarMovieItemProps> = ({ movie }) => {
  const classes = useStyles();
  return (
    <Link href={`/movies/${movie.movie_id}`} key={movie.movie_id}>
      <a className={classes.link}>
        <ListItem className={classes.similarMovieItem} key={movie.movie_id}>
          <Paper
            className={classes.similarMovieItemPaper}
            variant="outlined"
            square
          >
            <Grid container>
              <Grid item xs={2}>
                <Image src={movie.poster_url} aspectRatio={9 / 12} />
              </Grid>
              <Grid item xs={10}>
                <Typography className={classes.similarMovieTitle} variant="h6">
                  {makeTitle(movie)}
                </Typography>
              </Grid>
            </Grid>
          </Paper>
        </ListItem>
      </a>
    </Link>
  );
};

export default SimilarMovieItem;
