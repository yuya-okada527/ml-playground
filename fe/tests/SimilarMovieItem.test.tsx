import React from "react";
import { shallow } from "enzyme";
import SimilarMovieItem from "../src/components/SimilarMovieItem";
import { emptyMovie } from "./TestUtil";

describe("SimilarMovieItem", () => {
  it("Snapshot Test", () => {
    const movie = emptyMovie();
    const similarMovieItem = shallow(<SimilarMovieItem movie={movie} />);
    expect(similarMovieItem).toMatchSnapshot();
  });
});
