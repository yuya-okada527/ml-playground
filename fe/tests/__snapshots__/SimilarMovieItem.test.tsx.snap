// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimilarMovieItem Snapshot Test 1`] = `
<Link
  href="/movies/0"
  key="0"
>
  <a
    className="makeStyles-link-1"
  >
    <WithStyles(ForwardRef(ListItem))
      className="makeStyles-similarMovieItem-2"
      key="0"
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-similarMovieItemPaper-3"
        square={true}
        variant="outlined"
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={2}
          >
            <Image
              animationDuration={3000}
              aspectRatio={0.75}
              color="#fff"
              disableError={false}
              disableSpinner={false}
              disableTransition={false}
              errorIcon={
                <Memo
                  style={
                    Object {
                      "color": "#e0e0e0",
                      "height": 48,
                      "width": 48,
                    }
                  }
                />
              }
              loading={
                <WithStyles(ForwardRef(CircularProgress))
                  size={48}
                />
              }
              src=""
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={10}
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-similarMovieTitle-4"
              variant="h6"
            >
              申し訳ありません。タイトルが表示できません。
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(ListItem))>
  </a>
</Link>
`;
